### PR TITLE
feat: convert most Buffer usage to Uint8Array

### PR DESCRIPTION
**Pull request by bjornstar**
As you mentioned in https://github.com/Borewit/token-types/pull/650 a good test is if we can use `token-types` (& `file-type`) in `music-metadata`.

This is not exhaustive, there are still some Buffer usages remaining. However I have all the tests passing which I hope is good enough for a first pass.

I also discovered a race condition with the file-type parsing, we need to wait until the file type is parsed before we add the tags. 

This PR depends on changes to:

- [x] [`uint8array-extras`](https://github.com/sindresorhus/uint8array-extras/pull/12)
- [x] [`strtok3`](https://github.com/Borewit/strtok3/pull/1074)
- [x] [`token-types`](https://github.com/Borewit/token-types/pull/650)
- [x] [`file-type`](https://github.com/sindresorhus/file-type/pull/633)

### CODE REVIEW

**Comment by Borewit** [comment]
I propose the primary objective (requirement) is to ensure music-metadata can depend on [strtok3](https://github.com/Borewit/strtok3/pull/1074) and [token-type](https://github.com/Borewit/token-types/pull/650) the "_Goodbye Node.js Buffer_" versions. 

To replace remaining (all) Buffer dependencies in music-metadata is a secondary objective, which can be address in following stage (different PR).

The first objective, I think it is good to do before applying the changes in file-type (as health check as proposed [here](https://github.com/Borewit/token-types/pull/650#issuecomment-2196559072)). The remaining Buffer dependencies in music-metadata should not delay your effort to work towards getting this into file-type.

**Comment by bjornstar** [comment]
> I propose the primary objective (requirement) is to ensure music-metadata can depend on [strtok3](https://github.com/Borewit/strtok3/pull/1074) and [token-type](https://github.com/Borewit/token-types/pull/650), the "_Goodbye Node.js Buffer_" versions.
> 
> To replace remaining (all) Buffer dependencies in music-metadata is a secondary objective, which can be address in following stage (different PR).
> 
> The first objective, I think it is good to do before applying the changes in file-type (as health check as proposed [here](https://github.com/Borewit/token-types/pull/650#issuecomment-2196559072)). The remaining Buffer dependencies in music-metadata should not delay your effort to work towards getting this into file-type.
> 
> Only the secondary stage depends on the "_Goodbye Node.js Buffer_" version of [file-type](https://github.com/sindresorhus/file-type/pull/633).

I agree, we don't need to replace all the Buffer usage in the first pass. I think `file-type` can go first, but it should probably work  in either order.

Once the shared dependencies are released it should be clear that `file-type` and `music-metadata` both work with the new versions.

**Comment by Borewit** [comment]
Can you have a look to the remaining test cases, why these are failing?

**Comment by Borewit** [comment]
I managed to music-metadata pass all unit tests 💯✅, combining all outstanding PR's:
- [`strtok3`](https://github.com/Borewit/strtok3/pull/1074)
- [`token-types`](https://github.com/Borewit/token-types/pull/650)
- [`file-type`](https://github.com/sindresorhus/file-type/pull/633) (after fixing https://github.com/sindresorhus/file-type/pull/633/commits/4caaee2cb6a13425e528e00ec7387b6195ce277a)
- [this one: `music-metadata`](https://github.com/Borewit/music-metadata)

File: lib/id3v2/ID3v24TagMapper.ts [hunk 1]
```diff
@@ -190,7 +190,7 @@ export class ID3v24TagMapper extends CaseInsensitiveTagMap {
           case 'AverageLevel':
           case 'PeakValue':
             tag.id += ':' + tag.value.owner_identifier;
-            tag.value = tag.value.data.length === 4 ? tag.value.data.readUInt32LE(0) : null;
+            tag.value = tag.value.data.length === 4 ? new DataView(tag.value.data.buffer).getUint32(0) : null;
```
Comment (bjornstar) [inline lib/id3v2/ID3v24TagMapper.ts:193]: The original read was little endian so you need to pass `true` to `getUint32`.

```suggestion
            tag.value = tag.value.data.length === 4 ? new DataView(tag.value.data.buffer).getUint32(0, true) : null;
```

**Comment by bjornstar** [comment]
Appveyor is failing because `uint8array-extras` specifies node 18 as the minimum node version.

I added a fix for it in #2105

**Comment by bjornstar** [comment]
@Borewit This needs a new release of `strtok3` that includes https://github.com/Borewit/strtok3/pull/1074

**Comment by bjornstar** [comment]
All dependencies have been updated, this should be good to go.

**Comment by Borewit** [comment]
> All dependencies have been updated, this should be good to go.

Sorry for not being more responsive, my time is limited and I first objective is to finalize https://github.com/sindresorhus/file-type/pull/635, aiming release no Node.js dependent for file-type using the primary (non Node) entry point.

**Review by Borewit — APPROVED** [review]
Thanks a lot for your commitment and hard work @bjornstar !

**Comment by Borewit** [comment]
Part of [v9.0.0](https://github.com/Borewit/music-metadata/releases/tag/v9.0.0)

**Pull request merged**